libraries/AP_Math/AP_Math.cpp

```diff
@@ -4,13 +4,11 @@
 template <class FloatOne, class FloatTwo>
 bool is_equal(const FloatOne v_1, const FloatTwo v_2)
 {
-    static_assert(std::is_arithmetic<FloatOne>::value, "template parameter not of type float or int");
-    static_assert(std::is_arithmetic<FloatTwo>::value, "template parameter not of type float or int");
+    static_assert(std::is_floating_point<FloatOne>::value, "template parameter not of type float");
```

Comment:
The intention behind it is that this function can deal with mixed types. 
After this change it will throw assert if you compare a float with an int.